From 3ff39dc3749f332adcfb8e62be20d046dbd51706 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 29 Sep 2014 12:59:36 +0200
Subject: Correct compiler flags for libedataserver-private.la

The compiler flags for .cpp files should not use the same warnings
as the .c files.

diff --git a/libedataserver/Makefile.am b/libedataserver/Makefile.am
index 9a154ba..aad5c5a 100644
--- a/libedataserver/Makefile.am
+++ b/libedataserver/Makefile.am
@@ -18,8 +18,7 @@ BUILT_SOURCES = $(ENUM_GENERATED)
 lib_LTLIBRARIES = libedataserver-1.2.la
 noinst_LTLIBRARIES = libedataserver-private.la
 
-libedataserver_1_2_la_CPPFLAGS = \
-	$(AM_CPPFLAGS) \
+SHARED_COMPILER_FLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_srcdir)/private \
 	-I$(top_builddir) \
@@ -40,6 +39,10 @@ libedataserver_1_2_la_CPPFLAGS = \
 	$(ICU_CFLAGS) \
 	$(NULL)
 
+libedataserver_1_2_la_CPPFLAGS = \
+	$(AM_CPPFLAGS) \
+	$(SHARED_COMPILER_FLAGS)
+
 libedataserver_1_2_la_SOURCES = \
 	$(BUILT_SOURCES) \
 	e-alphabet-index-private.h \
@@ -185,7 +188,10 @@ libedataserver_private_la_SOURCES = \
 	e-alphabet-index-private.cpp \
 	e-transliterator-private.cpp \
 	$(NULL)
-libedataserver_private_la_CPPFLAGS = $(libedataserver_1_2_la_CPPFLAGS)
+libedataserver_private_la_CPPFLAGS = \
+	$(AM_CXXFLAGS) \
+	$(SHARED_COMPILER_FLAGS)
+
 if OS_WIN32
 libedataserver_1_2_la_LIBADD += libedataserver-private.la -lstdc++
 else
-- 
cgit v0.10.1

